mmf/models/transformers/base.py

```diff
@@ -134,29 +122,48 @@ def get_optimizer_parameters(self, config):
                         parameters=parameters,
                         param_list=param_list,
                     )
-            else:
+            elif name != "heads":
```

Comment:
The current logic you have added would miss shared parameters in `encoders`. So, I would suggest that you move this line 112 and change `elif` to `if`. In the end, there would be two `if` (no elif) conditions, one is specifically for adding all parameters that are no in the heads to a set and other is for encoders.